akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -105,8 +105,8 @@ class StubbedActorContext[T](
       case Some(inbox) ⇒ inbox.ref == child
     }
   }
-  override def watch[U](other: ActorRef[U]): ActorRef[U] = other
-  override def unwatch[U](other: ActorRef[U]): ActorRef[U] = other
+  override def watch(other: ActorRef[_]): Unit = ()
```

Comment:
I made that change in #22696 and added a testcase for watching from the Java API, but actually I'm not sure I understand what makes it more usable with the `[U]` from Java - could you elaborate?